### PR TITLE
Fix wording for "concepts/overview/working-with-objects/kubernetes-objects/#required-fields"

### PR DESCRIPTION
**Pull request by alghe-global**
### Fix wording for "concepts/overview/working-with-objects/kubernetes-objects/#required-fields"

Reworded references to Pod details (`spec` field) so it makes better sense when read.

### CODE REVIEW

**Comment by tengqm** [comment]
I failed to see a valuable improvement from this PR. Can we rewrite the text so that it reads more smoothly?

**Comment by alghe-global** [comment]
I don't have permissions to merge. Who is responsible for that?

File: content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md [hunk 1]
```diff
@@ -83,9 +83,8 @@ In the `.yaml` file for the Kubernetes object you want to create, you'll need to
 
 The precise format of the object `spec` is different for every Kubernetes object, and contains nested fields specific to that object. The [Kubernetes API Reference](https://kubernetes.io/docs/reference/kubernetes-api/) can help you find the spec format for all of the objects you can create using Kubernetes.
 
-For example, the reference for Pod details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
-for a Pod in the API, and the reference for Deployment details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/deployment-v1/#DeploymentSpec) for Deployments.
-In those API reference pages you'll see mention of PodSpec and DeploymentSpec. These names are implementation details of the Golang code that Kubernetes uses to implement its API.
+For example, see [this `spec` field](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec) reference for Pod details, for a Pod in the API, and [this `spec` field](/docs/reference/kubernetes-api/workload-resources/deployment-v1/#DeploymentSpec) reference for Deployment details, for Deployments.
```
Comment (kbhawkey) [inline content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md:86]: 👋 @alghe-global . I think the text could be clearer. Is this correct:

For example, see the details of the [Pod `spec`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec) and the definition of the [Deployment `spec`](/docs/reference/kubernetes-api/workload-resources/deployment-v1/#DeploymentSpec) in the Kubernetes API reference. Kubernetes uses a resource's `spec` to implement its API.

**Comment by sftim** [comment]
I'm also not sure I understand how this change improves things. I'm open to being persuaded but would prefer to have that explanation available before we merge this.

**Comment by alghe-global** [comment]
> I'm also not sure I understand how this change improves things. I'm open to being persuaded but would prefer to have that explanation available before we merge this.

It can be seen by reading the original text: there's no link between the subject and the rest of the description.

```
For example, the reference for Pod details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
for a Pod in the API, and the reference for Deployment details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/deployment-v1/#DeploymentSpec) for Deployments.
```

breaking it down:

```
For example, the reference for Pod details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
for a Pod in the API
```

doesn't specify what action one should take with regards to these references: should one check them? see them? do what exactly?

----

the same applies for the other part:

```
 and the reference for Deployment details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/deployment-v1/#DeploymentSpec) for Deployments.
```

the details/specs are just mentioned without however indication as to what/why  they're mentioned - this makes for a confusing reading when going through the text

File: content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md (hunk 1 above)
Comment (sftim) [inline content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md:86]: How about this:

```suggestion
For example, see the [`spec` field](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
in the reference for Pod. For each Pod, the `.spec` field specifies the pod and its desired state
(such as the container image name for each container within that pod).

Another example of an object specification is the
[`spec` field](/docs/reference/kubernetes-api/workload-resources/stateful-set-v1/#StatefulSetSpec)
for the StatefulSet API. For StatefulSet, the `.spec` field specifies the StatefulSet and
its desired state.
Within the `.spec` of a StatefulSet is a [template](/docs/concepts/workloads/pods/#pod-templates)
for Pod objects. That template describes Pods that the StatefulSet controller will create in order to
satisfy the StatefulSet specification.

Different kinds of object can also have different `.status`; again, the API reference pages
detail the structure of that `.status` field, and its content for each different type of object.
```

I switched to StatefulSet to avoid needing to explain what a ReplicaSet is.

Comment (alghe-global) [reply content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md:86]: * I'd reword `in the reference for Pod.` (in the reference of what?)
* Next line (line no. 87) needs to be updated since we're now talking about `StatefulSet` instead of Deployment.

Comment (sftim) [reply content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md:86]: Please feel free to revise your PR in line with the suggestions I've made, @alghe-global

Comment (sftim) [reply content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md:86]: As for “see the `spec` field in the reference for Pod” - that makes sense to me. I've elided _documentation_; a longer form would be “see the `spec` field in the reference documentation for Pod”.

If you want to suggest other wording, that's also fine.

File: content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md [hunk 2]
```diff
@@ -83,10 +83,19 @@ In the `.yaml` file for the Kubernetes object you want to create, you'll need to
 
 The precise format of the object `spec` is different for every Kubernetes object, and contains nested fields specific to that object. The [Kubernetes API Reference](https://kubernetes.io/docs/reference/kubernetes-api/) can help you find the spec format for all of the objects you can create using Kubernetes.
 
-For example, the reference for Pod details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
-for a Pod in the API, and the reference for Deployment details the [`spec` field](/docs/reference/kubernetes-api/workload-resources/deployment-v1/#DeploymentSpec) for Deployments.
-In those API reference pages you'll see mention of PodSpec and DeploymentSpec. These names are implementation details of the Golang code that Kubernetes uses to implement its API.
-
+For example, see the [`spec` field](/docs/reference/kubernetes-api/workload-resources/pod-v1/#PodSpec)
+for the Pod API reference (implementation details of the Golang code that Kubernetes uses to implement its API).
```
Comment (sftim) [inline content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md:87]: This is an improvement, but it's also slightly misleading. The Pod API reference is not “implementation details of the Golang code that Kubernetes uses to implement its API”. If someone wanted to write their own node implementation - some already exist - then those implementations are expected conform to the Pod API (and the Node API, etc) regardless of what programming language they use.

The _implementation details of the Golang code that Kubernetes uses to implement its API_ refers to the fragment identifier being named `PodSpec`, and that you see UpperCamelCase field names in that page - even though in the API it's not called a PodSpec; it's really a field named `spec` for resources of type `Pod`. In the Golang code for the kubelet, scheduler, etc, you do see `PodSpec` in upper camel case.

**Review by sftim — COMMENTED** [review]
/lgtm

There's still a caveat, so I opted not to also approve this - @alghe-global would you have time to revise this further.

File: content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md (hunk 2 above)
Comment (alghe-global) [reply content/en/docs/concepts/overview/working-with-objects/kubernetes-objects.md:87]: I think the confusion comes from the subject being discussed (`spec` field) and the details of Pod API to which the parentheses refers to (unless I'm missing something). I'll remove the parentheses so there's no disambiguation.
